src/main/java/soconnect/commons/util/StringUtil.java

```diff
@@ -55,22 +54,54 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
     public static boolean containsKeywordsIgnoreCase(String sentence, String keywords) {
         requireAllNonNull(sentence, keywords);
 
-        List<String> preppedKeywords = Arrays.asList(keywords.toUpperCase().trim().split("\\s+"));
+        String preppedWord = keywords.trim();
+        checkArgument(!preppedWord.isEmpty(), "Word parameter cannot be empty");
+        List<String> preppedKeywords = Arrays.asList(preppedWord.toUpperCase().trim().split("\\s+"));
```

Comment:
```suggestion
        List<String> preppedKeywords = Arrays.asList(preppedWord.toUpperCase().split("\\s+"));
```
The `.trim()` seems to be unnecessary because you've already trimmed it in line 57?

Reply:
True haha